src/lib.rs

```diff
@@ -0,0 +1,420 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use codec::{Codec, Decode, Encode};
+use rstd::collections::btree_set::BTreeSet;
+use rstd::prelude::*;
+use runtime_primitives::traits::{MaybeSerialize, Member, One, SimpleArithmetic};
+use srml_support::{
+    decl_event, decl_module, decl_storage, dispatch, ensure, traits::Get, Parameter, StorageMap,
+    StorageValue,
+};
+use system::{self, ensure_signed};
+
+mod error_messages;
+mod mock;
+mod tests;
+
+use error_messages::*;
+
+type MaxLength = u32;
+
+type MaxNumber = u32;
+
+type MaxConsecutiveRepliesNumber = u16;
+
+/// The pallet's configuration trait.
+pub trait Trait: system::Trait {
+
+    /// The overarching event type.
+    type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+
+    //Security/configuration constraints
+    type PostTitleMaxLength: Get<MaxLength>;
+
+    type PostBodyMaxLength: Get<MaxLength>;
+
+    type ReplyMaxLength: Get<MaxLength>;
+
+    type PostsMaxNumber: Get<MaxNumber>;
+
+    type RepliesMaxNumber: Get<MaxNumber>;
+
+    type DirectRepliesMaxNumber: Get<MaxNumber>;
+
+    type ConsecutiveRepliesMaxNumber: Get<MaxConsecutiveRepliesNumber>;
+
+    /// Type of identifier for blogs.
+    type BlogId: Parameter
+        + Member
+        + SimpleArithmetic
+        + Codec
+        + Default
+        + Copy
+        + MaybeSerialize
+        + PartialEq;
+
+    /// Type of identifier for blog posts.
+    type PostId: Parameter
+        + Member
+        + SimpleArithmetic
+        + Codec
+        + Default
+        + Copy
+        + MaybeSerialize
+        + PartialEq;
+
+    /// Type of identifier for replies.
+    type ReplyId: Parameter
+        + Member
+        + SimpleArithmetic
+        + Codec
+        + Default
+        + Copy
+        + MaybeSerialize
+        + PartialEq;
+}
+
+#[derive(Encode, Decode, Default, Clone, PartialEq)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub struct Post {
+    title: Vec<u8>,
+    body: Vec<u8>,
+}
+
+impl Post {
+    fn new(title: Vec<u8>, body: Vec<u8>) -> Self {
+        Self { title, body }
+    }
+}
+
+// Blog`s pallet storage items.
+decl_storage! {
+    trait Store for Module<T: Trait> as BlogModule {
+
+        // Blog Ids set, associated with owner
+        BlogIds get(fn blog_ids_by_owner): map T::AccountId => Option<BTreeSet<T::BlogId>>;
+
+        // Post Ids set, associated with blog
+        BlogPostIds get(fn post_ids_by_blog_id): map T::BlogId => Option<BTreeSet<T::PostId>>;
+
+        // Root replies
+        BlogPostParentReplyIds get (fn post_root_reply_ids): map (T::BlogId, T::PostId) => Option<BTreeSet<T::ReplyId>>;
+
+        // Direct replies to reply
+        BlogPostChildReplyIds get (fn post_child_reply_ids): map (T::BlogId, T::PostId, T::ReplyId) => Option<BTreeSet<T::ReplyId>>;
+
+        BlogPost get(fn post_by_id): map (T::BlogId, T::PostId) => Option<Post>;
```

Comment:
- Rename to `PostById`.

- Why is the value type an `Option`?

Reply:
In case of key pair non existence we will get none instead of default case for this value. Though we can check existence before accessing, using corresponding method, getter will return default representation.

Reply:
Renaming done